dev/ci.js

```diff
@@ -0,0 +1,82 @@
+import { createServer } from 'https'
+import { readFileSync } from 'fs'
+import { spawn, spawnSync } from 'child_process'
+import { once } from 'events'
+
+import { sendResponse } from './mocks.js'
+
+/*
+  every hours:
+    - git fetch
+    - cleanup stale database files (older than a week ?)
+
+*/
+
+const read = async (stream) => {
+  stream.setEncoding('utf8')
+  const chunks = []
+  for await (const chunk of stream) chunks.push(chunk)
+  return chunks.join('')
+}
+
+const handleRequest = async (req, res, again) => {
+  const hash = req.url.split('/', 2)[1]
+  const version = req.headers.referer?.match(
+    /https:\/\/([a-z0-9]{8})\.platform-au0l\.pages\.dev/,
+  )?.[1]
```

Comment:
first I get the `hash` from the url and the `version` form the referer headers.
This allows to also ignore non conform request, or request from other domains